src/main/java/com/innobridge/ethpay/service/TransactionService.java

```diff
@@ -94,5 +93,35 @@ public Transaction createTransaction(String senderId,
         return savedTransaction;
     }
 
+    public Transaction acceptTransaction(String transactionId) {
+        // TODO Auto-generated method stub
+        // Get transaction
+        Transaction transaction = transactionRepository.findById(transactionId)
+                .orElseThrow(() -> new IllegalArgumentException("Transaction not found: " + transactionId));
+
+        accountService.updatePendingTransaction(transaction.senderId, transaction.sourceCurrency);
+
+        transaction.setStatus(TransactionStatus.FILLED);
+        transaction.setCompletedDate(new Date(System.currentTimeMillis()));
+        return transactionRepository.save(transaction);
+    }
+
+    public Transaction rejectSenderTransaction(String transactionId, String message) {
+        Transaction transaction = transactionRepository.findById(transactionId)
+                .orElseThrow(() -> new IllegalArgumentException("Transaction not found: " + transactionId));
+        accountService.updatePendingTransaction(transaction.senderId, transaction.sourceCurrency);
+        transaction.setStatus(TransactionStatus.CANCELLED);
+        transaction.setCompletedDate(new Date(System.currentTimeMillis()));
+        return transactionRepository.save(transaction);
+    }
+
+    public Transaction rejectReceiverTransaction(String transactionId, String message) {
```

Comment:
pass in receiver id, verify that the id in the transaction is the same as the receiver